### PR TITLE
`FeatureFormView` - `BarcodeScannerFormInput` support

### PR DESCRIPTION
**Pull request by dfeinzimer**
Ref Apollo 866

### CODE REVIEW

File: Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        HStack {
+            Text(value.isEmpty ? String.noValue : value)
+            Spacer()
+            if !value.isEmpty {
+                ClearButton {
+                    value.removeAll()
+                }
+            }
+            Image(systemName: "barcode.viewfinder")
+                .foregroundStyle(.tint)
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift:54]: While this matches the approved design, it conflicts with the accesory symbol on date elements.

| `.tint` | `.secondary` |
|---|---|
|  |  |

**Comment by dfeinzimer** [comment]
Ready to go again

**Comment by dfeinzimer** [comment]
Ready to go

File: Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift (hunk 1 above)
Comment (mhdostal) [reply Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift]: I think they should match. @dfeinzimer @kaushikrw - any preference as to blue or gray (.tint or .secondary)?

Comment (kaushikrw) [reply Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift]: I think the gray looks uniform. On kotlin, the tint is either gray/white based on light/dark theme.

Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift]: Secondary/gray is my preference too. The microphone in other system applications is commonly gray.

| Maps | Contacts |
|---|---|
|  |  |

Comment (mhdostal) [reply Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift]: Ok, Gray it is. I'll update the design doc.

Comment (mhdostal) [reply Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift]: I've updated the design doc images.

**Review by mhdostal — APPROVED** [review]
Looks great and works really well. The button image difference between the flashlight "on" and "off" is very subtle; I missed it until I went into SF Symbols and saw the difference.

**Pull request merged**